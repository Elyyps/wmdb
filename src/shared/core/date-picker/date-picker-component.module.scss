.date-picker {
  margin-top: 24px;
}
.inputField {
  background-color: $global-background;
  height: 45px;
  width: 100%;
  font-size: 16px;
  padding: 10px 32px 10px 16px;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid $gray-lighter;
  font-family: $global-font-family;
  box-shadow: inset 0 1px 2px 0 rgba($black, 0.2);
  &:focus {
    outline: none;
    border-color: $global-color;
  }
}
label {
  font-size: 16px;
  font-weight: 600;
  margin-bottom: 8px;
  color: $global-color;
  display: block;
}
.popperClassName,
.calendarClassName {
  font-family: Assistant, sans-serif !important;
}

.calendarClassName {
  border-radius: 2px !important;
  border: 1px solid gray !important;
}

.datePickerWrapper {
  width: 100%;
  display: flex;
  > div:first-child {
    display: flex;
    width: 100%;
  }
}
