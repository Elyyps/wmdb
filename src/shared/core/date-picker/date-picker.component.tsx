import * as React from "react";
import styles from "./date-picker-component.module.scss";
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";

export interface IDatePickerComponentProps {
  errorMessage?: string;
  id: string;
  includeTime?: boolean;
  label?: string;
  onBlur: (date: Date | null, id: string) => void;
  onChange: (date: Date | null, id: string) => void;
  placeholder?: string | true;
  tooltip?: string;
  value?: string | null;
}

const DatePickerComponent = (props: IDatePickerComponentProps) => {
  const [startDate, setStartDate] = React.useState<Date | null>(null);

  const placeholderText =
    props.placeholder && typeof props.placeholder !== "string" ? "Voeg een datum en tijd toe" : props.placeholder;

  React.useEffect(() => {
    props.value ? setStartDate(new Date(props.value)) : setStartDate(null);
  }, [props.value]);

  const setChangeId = (date: Date | null, event: React.SyntheticEvent<any> | undefined) => {
    props.onChange(date, props.id);
  };

  return (
    <div className={styles["date-picker"]}>
      {props.label && <label htmlFor="">{props.label}</label>}
      <div className={styles.datePickerWrapper}>
        <DatePicker
          isClearable
          placeholderText={placeholderText}
          className={styles.inputField}
          popperClassName={styles.popperClassName}
          calendarClassName={styles.calendarClassName}
          selected={startDate}
          onChange={setChangeId}
          showTimeInput={props.includeTime}
          dateFormat="dd MMM yyyy - HH:mm"
          locale="nl"
          id={props.id}
        />
      </div>
    </div>
  );
};

export { DatePickerComponent };
