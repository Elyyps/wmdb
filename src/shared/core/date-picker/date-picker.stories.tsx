import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { DatePickerComponent } from "./date-picker.component";

storiesOf("DatePicker", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <DatePickerComponent onBlur={() => ""} onChange={() => ""} id={""} />);
