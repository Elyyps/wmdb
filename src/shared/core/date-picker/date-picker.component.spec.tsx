import React from "react";
import { shallow } from "enzyme";
import { DatePickerComponent } from "./date-picker.component";

describe("[DatePicker]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<DatePickerComponent onBlur={() => ""} onChange={() => ""} id={""} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
