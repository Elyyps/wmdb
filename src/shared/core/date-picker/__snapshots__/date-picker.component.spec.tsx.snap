// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[DatePicker] should render component without crashing 1`] = `
<div
  className="date-picker"
>
  <div
    className="datePickerWrapper"
  >
    <t
      allowSameDay={false}
      calendarClassName="calendarClassName"
      className="inputField"
      dateFormat="dd MMM yyyy - HH:mm"
      dateFormatCalendar="LLLL yyyy"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      id=""
      inlineFocusSelectedMonth={false}
      isClearable={true}
      locale="nl"
      monthsShown={1}
      nextMonthButtonLabel="Next Month"
      nextYearButtonLabel="Next Year"
      onBlur={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onInputClick={[Function]}
      onInputError={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      popperClassName="popperClassName"
      preventOpenOnFocus={false}
      previousMonthButtonLabel="Previous Month"
      previousYearButtonLabel="Previous Year"
      readOnly={false}
      renderDayContents={[Function]}
      selected={null}
      shouldCloseOnSelect={true}
      showMonthYearPicker={false}
      showQuarterYearPicker={false}
      showTimeInput={false}
      showTimeSelect={false}
      strictParsing={false}
      timeCaption="Time"
      timeInputLabel="Time"
      timeIntervals={30}
      withPortal={false}
    />
  </div>
</div>
`;
